.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,194 @@
+corresponding server release, the ml-develop branch reflects the latest changes
+that will be released next and are therefore the right place to commit changes.
+To be clear, the following branches have the assigned meaning:
+* ml-develop - the place to make changes for the next release of the MarkLogic NiFi Bundle
+* ml-release  - the stable code from the last release of the MarkLogic NiFi Bundle
+
+#### Create a branch for your changes
+
+Okay, so you have decided to fix something. Create a feature branch
+and start hacking:
+
+    $ git checkout -b my-feature-branch -t origin/ml-develop
+
+**NOTE: All work for pull requests to this fork should be restricted to the `<NIFI_PROJECT>/nifi-nar-bundles/nifi-marklogic-bundle/` folder. Any other work should be submitted directly to the official Apache NiFi repository.**
+
+#### Commit your changes
+
+Make sure git knows your name and email address:
+
+    $ git config --global user.name "J. Random User"
+    $ git config --global user.email "j.random.user@example.com"
+
+Writing good commit logs is important. A commit log should describe what
+changed and why. Follow these guidelines when writing one:
+
+1. The first line should be 50 characters or less and contain a short
+   description of the change including the Issue number prefixed by a hash (#).
+2. Keep the second line blank.
+3. Wrap all other lines at 72 columns.
+
+A good commit log looks like this:
+
+```
+Fixing Issue #123: make the whatchamajigger work in MarkLogic 8
+
+Body of commit message is a few lines of text, explaining things
+in more detail, possibly giving some background about the issue
+being fixed, etc etc.
+
+The body of the commit message can be several paragraphs, and
+please do proper word-wrap and keep columns shorter than about
+72 characters or so. That way `git log` will show things
+nicely even when it is indented.
+```
+
+The header line should be meaningful; it is what other people see when they
+run `git shortlog` or `git log --oneline`.
+
+#### Rebase your repo
+
+Use `git rebase` (not `git merge`) to sync your work from time to time.
+
+    $ git fetch upstream
+    $ git rebase upstream/ml-develop
+
+#### Test your code
+
+Be sure to run the tests before submitting your pull request. PRs with failing
+tests won't be accepted.
+
+First verify that you have valid server and admin auth info in
+`username` and `password` fields in
+`nifi-nar-bundles/nifi-marklogic-bundle/nifi-marklogic-processors/src/test/java/org/apache/nifi/marklogic/processor/AbstractMarkLogicIT.java` and `nifi-nar-bundles/nifi-marklogic-bundle/nifi-marklogic-services/src/test/java/org/apache/nifi/marklogic/controller/DefaultMarkLogicDatabaseClientServiceIT.java`
+
+If compatible NiFi SNAPSHOT hasn't been installed in your local Maven yet (Grab a cup of coffee while it runs.):
+
+    $ mvn -T 2 clean install -Pcontrib-check,include-grpc,include-atlas,include-hive3 -Ddir-only | grep -v -F -f .travis-output-filters
```

Comment:
if you're running on Windows, I don't think this will work unless you add the skip-tests option

Reply:
```suggestion
    $ mvn -T 2 clean install -Ddir-only -DskipTests | grep -v -F -f .travis-output-filters
```